eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -56,6 +63,24 @@ case class RouterConf(randomizeRouteSelection: Boolean,
                       searchRatioChannelCapacity: Double)
 
 case class ChannelDesc(shortChannelId: ShortChannelId, a: PublicKey, b: PublicKey)
+
+case class PublicChannel(ann: ChannelAnnouncement, fundingTxid: ByteVector32, capacity: Satoshi, update_1_opt: Option[ChannelUpdate], update_2_opt: Option[ChannelUpdate]) {
+  update_1_opt.foreach(u => assert(Announcements.isNode1(u.channelFlags)))
+  update_2_opt.foreach(u => assert(!Announcements.isNode1(u.channelFlags)))
+  def getNodeIdSameSideAs(u: ChannelUpdate): PublicKey = if (Announcements.isNode1(u.channelFlags)) ann.nodeId1 else ann.nodeId2
+  def getSameSideAs(u: ChannelUpdate): Option[ChannelUpdate] = if (Announcements.isNode1(u.channelFlags)) update_1_opt else update_2_opt
+  def updateSameSideAs(u: ChannelUpdate): PublicChannel = if (Announcements.isNode1(u.channelFlags)) copy(update_1_opt = Some(u)) else copy(update_2_opt = Some(u))
+  def updateFor(n: PublicKey): Option[ChannelUpdate] = if (n == ann.nodeId1) update_1_opt else if (n == ann.nodeId2) update_2_opt else throw new IllegalArgumentException("this node is unrelated to this channel")
```

Comment:
Naming here is confusing,  in `updateSameSideAs` update means "modify", here it means "get" , why not simply call it `get` which would be consistent with  `getSameSideAs` ?